stories/demos/HeaderFilters.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { useMemo, useState } from 'react';
+import { AutoSizer } from 'react-virtualized';
+import Select from 'react-select';
+import faker from 'faker';
+
+import DataGrid, { Column, Filters } from '../../src';
+import { NumericFilter } from './components/Filters';
+import Toolbar from './components/Toolbar/Toolbar';
+
+interface Row {
+  id: number;
+  task: string;
+  priority: string;
+  issueType: string;
+  developer: string;
+  complete: number;
+}
+
+function createRows() {
+  const rows: Row[] = [];
+  for (let i = 1; i < 500; i++) {
+    rows.push({
+      id: i,
+      task: `Task ${i}`,
+      complete: Math.min(100, Math.round(Math.random() * 110)),
+      priority: ['Critical', 'High', 'Medium', 'Low'][Math.floor(Math.random() * 4)],
+      issueType: ['Bug', 'Improvement', 'Epic', 'Story'][Math.floor(Math.random() * 4)],
+      developer: faker.name.findName()
+    });
+  }
+  return rows;
+}
+
+export default function HeaderFilters() {
+  const [rows] = useState(createRows);
+  const [filters, setFilters] = useState<Filters<Row>>({
+    priority: 'Critical',
+    issueType: ''
+  });
+  const [filteredRows, setFilteredRows] = useState(() => filterRows(filters));
```

Comment:
For derived states we should use `useMemo` instead of a state. It's so much safer IMO.
Example: https://github.com/adazzle/react-data-grid/blob/canary/stories/demos/CommonFeatures.tsx#L204

Reply:
I completely agree 👍